### PR TITLE
feat: Adds configurable skill and stat gain

### PR DESCRIPTION
**Pull request by thecatontheflat**
This addresses the following issue: https://github.com/modernuo/ModernUO/issues/1486

Changes:

- Added the following configuration parameters:
   - `skills.gainValueMultiplier` - multiplies the value of the gained Skill
   - `stats.gainValueMultiplier` - multiplies the value of the gained Stat
   - `stats.gainChanceMultiplier` - multiplies the chance of a Stat gain to proc
- Changes to the logic:
   - Added checks to make sure that gain/loss does not exceed the caps nor goes below 0
   - When a Skill has a chance of increasing multiple Stats, all of them will make a roll now (previously it would stop at a first successful roll)
   - Made conditions short-circuit on a simpler condition first before triggering the random number generator

image

### CODE REVIEW

File: Projects/UOContent/Skills/SkillCheck.cs [hunk 1]
```diff
@@ -299,17 +347,17 @@ public static void IncreaseStat(Mobile from, Stat stat, bool atrophy)
                     {
                         if (CanLower(from, Stat.Dex) && (from.RawDex < from.RawInt || !CanLower(from, Stat.Int)))
                         {
-                            --from.RawDex;
+                            from.RawDex = DecreaseStatOrZero(from.RawDex);
```
Comment (kamronbatman) [inline Projects/UOContent/Skills/SkillCheck.cs:350]: The stat property itself already has these checks. So we don't need to do them here.

**Comment by thecatontheflat** [comment]
@kamronbatman It would be best if you take it over, I won't be able to contribute in the next few weeks

**Comment by kamronbatman** [comment]
@thecatontheflat
- Added the 75/25 Primary/Secondary stat for Publish 45 with its stat gain mechanics.
- Removed the stat/skill value gain multipliers since they were ruining the ability to fine-tune skills and decrease/increase accordingly.
- Simplified the logic to prevent going over the gain cap.
- Reduced the stat delay from 15 minutes to 10 minutes per OSI.

Note: I did not add the 15 stats per day limit cause nobody wants a slow game these days and it was removed in UOML.

**Comment by thecatontheflat** [comment]
Thank you!

**Pull request merged**